packages/authentication/src/strategy-adapter.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import * as http from 'http';
+import {ServerRequest as Request} from 'http';
+import {HttpErrors, ParsedRequest} from '@loopback/core';
+
+/**
+ * Interface definition of a passport strategy.
+ */
+export interface Strategy {
+  authenticate(req: http.ServerRequest): void;
```

Comment:
Don't we expect a Promise of the result? Typically, authentication will either fail or establish a user.